### ISSUE TITLE
KBFS sync stuck

### ISSUE THREAD

**Issue by piotr-szegda** [keybase/client#22816]
I'd like to switch to keybase KBFS as main cloud drive. I copied files from previous cloud to keybase folder. I also copied old backup of phone storage with a lot of small files. That made KBFS slow. I realized that I don't need this files so I stopped copy and tried to delete it via cli. Now KBFS stuck and stop sending files.

It's seem to stop on endless loop:
```
2020-02-29T11:04:43.010876+01:00 ▶ [DEBU kbfs(PRE) prefetcher.go:1036] 3429 {VDL:2} Handling request for BlockPointer{ID: 02dfc33df7d31555b8529a47a283fb408b40bfc56d4c6211a20614569050a317d6, KeyGen: 6, DataVer: 1, Context: Context{Creator: 6bc8f06a40c7d63bdcf0311ef45ec819}, DirectType: direct}, action=prefetch
2020-02-29T11:04:43.010909+01:00 ▶ [DEBU kbfs(PRE) prefetcher.go:1104] 342a {VDL:2} skipping prefetch for finished block 02dfc33df7d31555b8529a47a283fb408b40bfc56d4c6211a20614569050a317d6
2020-02-29T11:04:43.086114+01:00 ▶ [DEBU kbfs(FBO ffdcac22) folder_branch_ops.go:6268] 342b [duration=2.694003352s] Status done: <nil> [tags:SFSID=lUyXJ9LgeRfGxstF9V2wKQ]
2020-02-29T11:04:43.086507+01:00 ▶ [DEBU kbfs(simplefs) simplefs.go:2308] 342c Sending sync status response with 97120560891 syncing bytes [tags:SFSID=lUyXJ9LgeRfGxstF9V2wKQ]
2020-02-29T11:04:47.286559+01:00 ▶ [DEBU kbfs(FBO ffdcac22) folder_branch_ops.go:6258] 342d {VDL:1} Status [tags:SFSID=2tbawjsK5qnEolLMRseJvA]
2020-02-29T11:04:47.286661+01:00 ▶ [DEBU kbfs(ECQU - KBFSOps) fetch_decider.go:135] 342e {VDL:1} Cached data is 6.753956554s old [tags:SFSID=2tbawjsK5qnEolLMRseJvA]
2020-02-29T11:04:47.286682+01:00 ▶ [DEBU kbfs(ECQU - KBFSOps) fetch_decider.go:70] 342f {VDL:1} Spawning fetch in background with tag:ECQU=WSW7xxyTjEm7_0a_RzSaPA [tags:SFSID=2tbawjsK5qnEolLMRseJvA]
2020-02-29T11:04:47.286861+01:00 ▶ [DEBU kbfs(ECQU - KBFSOps) quota_usage.go:208] 3430 Read quota for 6bc8f06a40c7d63bdcf0311ef45ec819 from disk cache [tags:ECQU=WSW7xxyTjEm7_0a_RzSaPA]
2020-02-29T11:04:47.372606+01:00 ▶ [DEBU kbfs(TLFJ) tlf_journal.go:1868] 3431 Making incomplete unflushed path cache [tags:SFSID=2tbawjsK5qnEolLMRseJvA]
2020-02-29T11:04:47.427876+01:00 ▶ [DEBU kbfs(ECQU - KBFSOps) quota_usage.go:193] 3432 getAndCache: error=<nil> [tags:ECQU=WSW7xxyTjEm7_0a_RzSaPA]
2020-02-29T11:04:47.427912+01:00 ▶ [DEBU kbfs(ECQU - KBFSOps) fetch_decider.go:92] 3433 {VDL:1} Finished fetch: <nil> [tags:ECQU=WSW7xxyTjEm7_0a_RzSaPA]
2020-02-29T11:04:48.286732+01:00 ▶ [DEBU kbfs(FBO ffdcac22) asm_amd64.s:1357] 3434 (Long operation, started=2020-02-29 11:04:47.286542315 +0100 CET m=+4569.279151433) Status [tags:SFSID=2tbawjsK5qnEolLMRseJvA]
2020-02-29T11:04:49.499637+01:00 ▶ [DEBU kbfs(FBO ffdcac22) folder_block_ops.go:3330] 3435 {VDL:1} Root node BlockPointer{ID: 02549784ad93cc353004806d1b733359bd534871f6af6dbbe30706384226a75ccf, KeyGen: 6, DataVer: 1, Context: Context{Creator: 6bc8f06a40c7d63bdcf0311ef45ec819}, DirectType: direct} doesn't exist in the node cache; using a throwaway node cache instead [tags:SFSID=2tbawjsK5qnEolLMRseJvA]
2020-02-29T11:04:49.520092+01:00 ▶ [DEBU kbfs(PRE) prefetcher.go:1036] 3436 {VDL:2} Handling request for BlockPointer{ID: 02dfc33df7d31555b8529a47a283fb408b40bfc56d4c6211a20614569050a317d6, KeyGen: 6, DataVer: 1, Context: Context{Creator: 6bc8f06a40c7d63bdcf0311ef45ec819}, DirectType: direct}, action=prefetch
2020-02-29T11:04:49.520164+01:00 ▶ [DEBU kbfs(PRE) prefetcher.go:1104] 3437 {VDL:2} skipping prefetch for finished block 02dfc33df7d31555b8529a47a283fb408b40bfc56d4c6211a20614569050a317d6

...
After long time
2020-02-29T11:51:12.209256+01:00 ▶ [WARN kbfs folder_branch_status.go:341] 4c60 Error getting journal status for ffdcac227b772889a80cd7eb80787416: context deadline exceeded [tags:SFSID=_jLQdx5ZczKaoInib8uw_Q]
2020-02-29T11:51:12.209288+01:00 ▶ [DEBU kbfs(FBO ffdcac22) folder_branch_ops.go:6268] 4c61 [duration=5.060301174s] Status done: <nil> [tags:SFSID=_jLQdx5ZczKaoInib8uw_Q]
2020-02-29T11:51:12.209318+01:00 ▶ [DEBU kbfs(simplefs) simplefs.go:2308] 4c62 Sending sync status response with 0 syncing bytes [tags:SFSID=_jLQdx5ZczKaoInib8uw_Q]

and nothing more.

```

private/piotrszegda/.kbfs_status is very long so I just put beginning of file. Rest is very repetitive. List probably contains deleted file that I mentioned before. 
```
{
  "Staged": true,
  "BranchID": "6459fd100e27e2acf373f2645e97c1a3",
  "HeadWriter": "piotrszegda",
  "DiskUsage": 99189564739,
  "RekeyPending": false,
  "LatestKeyGeneration": 6,
  "FolderID": "ffdcac227b772889a80cd7eb80787416",
  "Revision": 4124,
  "LastGCRevision": 75,
  "MDVersion": 4,
  "RootBlockID": "02607224bb7d159842499a7d37e10fa926a20c3b7371ced69c1f11226648dd4997",
  "SyncEnabled": false,
  "PrefetchStatus": "NoPrefetch",
  "UsageBytes": 16940196004,
  "ArchiveBytes": 568028544,
  "LimitBytes": 268435456000,
  "GitUsageBytes": 4035928,
  "GitArchiveBytes": 43238,
  "GitLimitBytes": 107374182400,
  "LocalTimestamp": "2020-02-29T07:38:11.955682016+01:00",
  "DirtyPaths": null,
  "Unmerged": [
    {
      "Path": "Unknown path: BlockPointer{ID: 02f31eed8c0801c9e374aab19029862efb3b7f72b7467944c69365e63589c2b92c, KeyGen: 6, DataVer: 1, Context: Context{Creator: 6bc8f06a40c7d63bdcf0311ef45ec819}, DirectType: direct}",
      "Ops": [
        "setAttr miracle-light (mtime)",
        "setAttr miracle-light (mtime)",
        "setAttr miracle-light (mtime)"
      ]
    },
    {
      "Path": "Unknown path: BlockPointer{ID: 02bda4c44799b54b72a0ce3790c00e2022294235b4411fc1c937e82b7c3ca793a9, KeyGen: 6, DataVer: 1, Context: Context{Creator: 6bc8f06a40c7d63bdcf0311ef45ec819}, DirectType: direct}",
      "Ops": [
        "setAttr worry-weather.pdf (mtime)",
        "sync [{off=0, len=405539}, {off=405539, len=0}]"
      ]
    },
    {
      "Path": "Unknown path: BlockPointer{ID: 02e20aa982cd22d1b24c4bff4dd4702ad69a35a8ed7a1eedd8828fe7af248e428f, KeyGen: 6, DataVer: 1, Context: Context{Creator: 6bc8f06a40c7d63bdcf0311ef45ec819}, DirectType: indirect}",
      "Ops": [
        "setAttr zero-desk.jpg (mtime)",
        "sync [{off=0, len=1082258}, {off=1082258, len=0}]"
      ]
    },
    {
      "Path": "Unknown path: BlockPointer{ID: 0240fcd413d1acc41f35f25382f1238f36388485e572d8c89b4a7a2284390a79c0, KeyGen: 6, DataVer: 1, Context: Context{Creator: 6bc8f06a40c7d63bdcf0311ef45ec819}, DirectType: indirect}",
      "Ops": [
        "setAttr crash-tackle.jpg (mtime)",
        "sync [{off=0, len=539437}, {off=539437, len=0}]"
      ]
    },
    {

```

my log id: 3facfd8b6d16f3f09773a01c

**Comment by strib** [comment]
Unfortunately this KBFS log is not included in your log send.  Not sure why -- are you running Keybase in an unusual way?  Hard for me to debug for sure without the logs.  If you have them in a separate file, maybe you can share them with me another way, like in a shared folder on KBFS if the file system is working?

It seems your local device got into a conflict, perhaps because a different device wrote to the folder at the same time.  Now it's trying to resolve the conflict, but you have so much data that there it might be choking on the large amount of data, or perhaps it hit some kind of deadlock.  Please let me know if this is still happening, and if there's any way you can get me the full KBFS log you were looking at.

**Comment by piotr-szegda** [comment]
Two days ago I tried send file from other device (laptop). KBFS on laptop used more then 16GB of RAM but after whole night sync ended successfully. After that I started KBFS on stuck device and after few hours I received message about conflict so I deleted local version and sync finally started working.

On stuck device I am using NixOS so it's pretty unusual distro and after stuck I updated from 5.1.1 to last version.

For me problem it's solved but if log files will be helpful I am gladly to upload it.
